src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,541 @@
+package org.labkey.test.components.ui.grids;
+
+import org.junit.Assert;
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.UpdatingComponent;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.html.Checkbox;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Wraps EntityFieldFilterModal.tsx in UI components.
+ */
+public class CustomizeGridDialog extends ModalDialog
+{
+    private final UpdatingComponent linkedComponent;
+
+    private static final String FIELD_NOT_AVAILABLE = "Field name '%s' is not visible in the 'Available Fields' list.";
+    private static final String FIELD_NOT_IN_GRID = "Field with label '%s' is not visible in the 'Shown in Grid' list.";
+
+    public CustomizeGridDialog(WebDriver driver, UpdatingComponent linkedComponent)
+    {
+        super(new ModalDialogFinder(driver));
+        this.linkedComponent = linkedComponent;
+        waitForReady();
+    }
+
+    @Override
+    protected void waitForReady()
+    {
+        WebDriverWrapper.waitFor(()->
+                {
+                    WebElement spinner = BootstrapLocators.loadingSpinner.findWhenNeeded(this);
+                    return !spinner.isDisplayed() &&
+                            elementCache().contentPanelLocator.findElements(this).size() == 2;
+                },
+                "Customize Grid dialog did not render in time.", 1_500);
+    }
+
+    /**
+     * Check or uncheck the 'Show all system and user-defined fields' checkbox checked?
+     *
+     * @param checked Set to true to check the box, false to uncheck it.
+     * @return True if checked false otherwise.
+     */
+    public CustomizeGridDialog setShowAll(boolean checked)
+    {
+        elementCache().checkbox.set(checked);
+        return this;
+    }
+
+    /**
+     * Is the 'Show all system and user-defined fields' checkbox checked?
+     *
+     * @return True if checked false otherwise.
+     */
+    public boolean isShowAllChecked()
+    {
+        return elementCache().checkbox.isChecked();
+    }
+
+    /**
+     * Get the list of visible fields from the 'Available Fields' panel. Children of an expanded filed only have the name
+     * in the list, this des not include the path.
+     *
+     * @return The list of field names.
+     */
+    public List<String> getAvailableFields()
+    {
+        List<WebElement> listItemElements = elementCache().getListItemNameElements(elementCache().availableFieldsPanel);
+        return listItemElements.stream().map(WebElement::getText).collect(Collectors.toList());
```

Comment:
IntelliJ always suggest simplifying usages of `Collectors.toList`
```suggestion
        return listItemElements.stream().map(WebElement::getText).toList();
```

Reply:
Yes, but that makes it an immutable list. I would stumble over that in tests that would like to add/remove elements from the list to use for comparisons.